x-pack/plugins/apm/public/context/charts_sync_context.tsx

```diff
@@ -23,10 +26,11 @@ export function ChartsSyncContextProvider({
   children: ReactNode;
 }) {
   const [event, setEvent] = useState({});
+  const { annotations } = useAnnotations();
```

Comment:
Not sure this is the right place... Do we want annotations on all charts?

Reply:
If this was how it was previously implemented, I'm good with this

Reply:
I thought a bit more about this, and I don't think annotations fit into the `ChartsSyncContext`. To make this clear [I opened a PR](https://github.com/elastic/kibana/pull/84272) that renames the `ChartsSyncContext` to `PointerEventContext`.

If we want to re-use annotations multiple places I suggest you create an AnnotationContext that retrieves the annotations. Each chart can get consume the annotations like they used to (via `useAnnotations`). The difference is that `useAnnotations` doesn't fetch the annotations so we shouldn't see multiple requests.

Reply:
> If we want to re-use annotations multiple places I suggest you create an AnnotationContext that retrieves the annotations. Each chart can get consume the annotations like they used to (via useAnnotations). The difference is that useAnnotations doesn't fetch the annotations so we shouldn't see multiple requests.

That sounds good too. I'm going to work on that.